packages/adminrouter/extra/src/test-harness/tests/test_service.py

```diff
@@ -0,0 +1,473 @@
+# Copyright (C) Mesosphere, Inc. See LICENSE file for details.
+
+import time
+
+import pytest
+import requests
+
+from generic_test_code.common import (
+    generic_correct_upstream_dest_test,
+    generic_correct_upstream_request_test,
+    header_is_absent
+)
+from mocker.endpoints.marathon import NGINX_APP_ALWAYSTHERE_DIFFERENTPORT
+from mocker.endpoints.mesos import (
+    NGINX_FWRK_ALWAYSTHERE_DIFFERENTPORT,
+    NGINX_FWRK_ALWAYSTHERE_NOWEBUI,
+    framework_from_template
+)
+from mocker.endpoints.mesos_dns import NGINX_SRV_ALWAYSTHERE_DIFFERENTPORT
+
+CACHE_UPDATE_DELAY = 2  # seconds
+assert CACHE_UPDATE_DELAY > 1.5  # due to cache_expiration=(CACHE_UPDATE_DELAY - 1)
+
+
+@pytest.fixture()
+def master_ar_process_fastcache(nginx_class):
+    """An AR process instance fixture for situations where a short lived cache
+       is needed. In order to achieve it, the scope of this fixture needs to be
+       single test.
+    """
+    nginx = nginx_class(role="master",
+                        cache_poll_period=CACHE_UPDATE_DELAY,
+                        cache_expiration=(CACHE_UPDATE_DELAY - 1),
+                        cache_first_poll_delay=1,
+                        )
+    nginx.start()
+
+    yield nginx
+
+    nginx.stop()
+
+
+class TestServiceStatefull:
+    # Test all the statefull test-cases/tests where AR caching may influence the
+    # results
+    def test_if_marathon_apps_are_resolved(
```

Comment:
It seems that second part of this test (after the `set_apps_response`) is checking same code path in `service.lua` and only thing that is different is the data yielded by cache module. Since the `service.lua` code is stateless I don't see value of second part, especially since we have cache code already tested.